qiskit_experiments/library/characterization/analysis/ramsey_xy_analysis.py

```diff
@@ -422,6 +429,69 @@ def _default_options(cls):
 
         return options
 
+    def _format_data(
+        self,
+        curve_data: curve.ScatterTable,
+    ) -> curve.ScatterTable:
+        columns = list(curve_data.columns)
+
+        formatted_data = super()._format_data(curve_data)
+        raw_ramsey = formatted_data.filter(like="formatted", axis="index")
+
+        # Convert to phase format with unwrapper
+        phase_data = np.empty((0, len(columns)))
+        y_mean = raw_ramsey.yval.mean()
+        grouped = raw_ramsey.groupby(["direction", "series"])
+
+        for m_id, direction in enumerate(("pos", "neg")):
+            x_quadrature = grouped.get_group((direction, "X"))
+            y_quadrature = grouped.get_group((direction, "Y"))
+            if not np.array_equal(x_quadrature.xval, y_quadrature.xval):
+                raise ValueError(
+                    "Amplitude values of X and Y quadrature are different. "
+                    "Same values must be used."
+                )
+            x_uarray = unp.uarray(x_quadrature.yval, x_quadrature.yerr)
+            y_uarray = unp.uarray(y_quadrature.yval, y_quadrature.yerr)
+
+            amplitudes = x_quadrature.xval.to_numpy()
+
+            # pylint: disable=no-member
+            phase = unp.arctan2(y_uarray - y_mean, x_uarray - y_mean)
+            phase_n = unp.nominal_values(phase)
+            phase_s = unp.std_devs(phase)
+
+            # Unwrap phase
+            # We assume a smooth slope and correct 2pi phase jump to minimize the change of the slope.
+            if amplitudes[0] < 0:
+                # Flip array order because this is array is negative increments.
+                phase_n = phase_n[::-1]
+            unwrapped_phase = np.unwrap(phase_n)
+            if amplitudes[0] < 0:
+                # Flip back
+                unwrapped_phase = unwrapped_phase[::-1]
+
+            # Store new data
+            tmp = np.empty((len(amplitudes), len(columns)), dtype=object)
+            tmp[:, columns.index("xval")] = amplitudes
+            tmp[:, columns.index("yval")] = unwrapped_phase
+            tmp[:, columns.index("yerr")] = phase_s
+            tmp[:, columns.index("model_name")] = f"F{direction}"
```

Comment:
F stands for frequency. There are `Xpos`, `Xneg`, `Ypos`, `Yneg`, `Fpos`, `Fneg` in the scatter table, which will be saved in artifact once we activate it.

Reply:
Okay, it might be worth mentioning in a docstring somewhere for clarity. Why "frequency" rather than "phase"?

Reply:
I choose frequency because this is the quantity of interest, but I realized that we can keep phase internally and convert it into frequency at the visualization step. This becomes θpos/neg now.

Reply:
Oh, right, the dependence of frequency on amplitude is complicated but the dependence of frequency on phase at constant amplitude is simple. Maybe it would be better to divide by `2 * pi * ts` in `_format_data` and store the frequency rather than the phase? That might be better for a user wanting to work with the scatter table after running the analysis.

Reply:
That makes sense :) done in [994642f](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1243/commits/994642f64fdafd152a6cea23aab93b90fd97ccde)